ember-primitives/src/components/accordion.gts

```diff
@@ -0,0 +1,168 @@
+   */
+  value: string[];
+  /**
+   * A callback that is called when the selected values change. To be used in a controlled fashion in conjunction with `value`.
+   */
+  onValueChange: (value?: string[] | undefined) => void;
+  /**
+   * Not available in a controlled fashion.
+   */
+  defaultValue?: never;
+} | {
+  /**
+   * Not available in an uncontrolled fashion.
+   */
+  value?: never;
+  /**
+   * Not available in an uncontrolled fashion.
+   */
+  onValueChange?: never;
+  /**
+   * The default values of the accordion. To be used in an uncontrolled fashion.
+   */
+  defaultValue?: string[];
+})
+
+export class Accordion extends Component<{
+  Element: HTMLDivElement;
+  Args: AccordionSingleArgs | AccordionMultipleArgs;
+  Blocks: {
+    default: [{
+      /**
+       * The AccordionItem component.
+       */
+      Item: WithBoundArgs<typeof AccordionItem, 'selectedValue' | 'toggleItem' | 'disabled'>
+    }];
+  };
+}> {
+  <template>
+    <div data-disabled={{@disabled}} ...attributes>
+      {{yield (hash Item=(component AccordionItem selectedValue=this.selectedValue toggleItem=this.toggleItem disabled=@disabled))}}
+    </div>
+  </template>
+
+  @localCopy('args.defaultValue') declare _internallyManagedValue?: string | string[];
+
+  get selectedValue() {
+    return this.args.value ?? this._internallyManagedValue;
+  }
+
+  toggleItem = (value: string) => {
+    if (this.args.disabled) {
+      return;
+    }
+
+    if (this.args.type === 'single') {
+      this.toggleItemSingle(value);
+    } else if (this.args.type === 'multiple') {
+      this.toggleItemMultiple(value);
+    }
+  }
+
+  toggleItemSingle = (value: string) => {
+    assert('Cannot call `toggleItemSingle` when `disabled` is true.', !this.args.disabled);
+    assert('Cannot call `toggleItemSingle` when `type` is not `single`.', this.args.type === 'single');
+
+    const newValue = value === this.selectedValue ? undefined : value;
+
+    if (this.args.onValueChange) {
+      this.args.onValueChange(newValue);
+    } else {
+      this._internallyManagedValue = newValue;
+    }
+  }
+
+  toggleItemMultiple = (value: string) => {
+    assert('Cannot call `toggleItemMultiple` when `disabled` is true.', !this.args.disabled);
+    assert('Cannot call `toggleItemMultiple` when `type` is not `multiple`.', this.args.type === 'multiple');
+
+    const currentValues = this.selectedValue as string[] | undefined ?? [];
+    const indexOfValue = currentValues.indexOf(value);
+    let newValue: string[];
+
+    if (indexOfValue === -1) {
+      newValue = [...currentValues, value];
+    } else {
+      newValue = [
+        ...currentValues.slice(0, indexOfValue),
+        ...currentValues.slice(indexOfValue + 1)
+      ];
+    }
+
+    if (this.args.onValueChange) {
+      this.args.onValueChange(newValue);
+    } else {
+      this._internallyManagedValue = newValue;
+    }
+  }
+}
+
+export default Accordion;
+
+export { AccordionContent } from './accordion/content';
```

Comment:
do these need to be exported?

Reply:
I think so, when I remove them (and also the re-export in `index.ts`) I can no longer get their API reference working in the docs page

Reply:
Since some of them have their arguments configured, i wonder if it'd be better to have separate types for those, and export the types for the pre-wired components?

Reply:
But also, this is something i can look in to later if you don't want to worry about. It is puzzling 🤔

Reply:
That would make more sense indeed! Currently it looks like you'd need to e.g.  pass `@disabled` to `AccordionContent` which isn't actually necessary as the parent component takes care of that.

I'll look into adding these separate types